web/src/components/storage/SpacePolicySelection.tsx

```diff
@@ -84,37 +82,27 @@ const SpacePolicyPicker = ({ currentPolicy, onChange = noop }) => {
  * Renders a page that allows the user to select the space policy and actions.
  */
 export default function SpacePolicySelection() {
-  const [state, setState] = useState({ load: false, settings: {} });
-  /** @type ReturnType<typeof useState<SpacePolicy>> */
-  const [policy, setPolicy] = useState();
+  const { settings } = useProposalResult();
+  const updateProposal = useProposalMutation();
```

Comment:
NP: just a note from my side for this and similar _mutation hooks_. A matter of taste, but I use to write them in a different way:

```suggestion
  const { mutateAsync: updateProposal } = useProposalMutation();
```

And then just use the `updateProposal(withWhatever)`.

But as said, just a **personal** preference.